packages/react-charting/src/components/GaugeChart/GaugeChart.base.tsx

```diff
@@ -0,0 +1,627 @@
+import * as React from 'react';
+import * as shape from 'd3-shape';
+import { classNamesFunction, getRTL } from '@fluentui/react/lib/Utilities';
+import {
+  IGaugeChartProps,
+  IGaugeChartSegment,
+  IGaugeChartStyleProps,
+  IGaugeChartStyles,
+  GaugeValueFormat,
+} from './GaugeChart.types';
+import { IProcessedStyleSet } from '@fluentui/react/lib/Styling';
+import {
+  Points,
+  convertToLocaleString,
+  formatValueWithSIPrefix,
+  getAccessibleDataObject,
+  getColorFromToken,
+  pointTypes,
+} from '../../utilities/index';
+import { ILegend, LegendShape, Legends, Shape } from '../Legends/index';
+import { FocusZone, FocusZoneDirection } from '@fluentui/react-focus';
+import { Callout, DirectionalHint } from '@fluentui/react/lib/Callout';
+import { IYValueHover } from '../../index';
+import { TooltipText } from '../../utilities/TooltipText';
+import { select as d3Select } from 'd3-selection';
+
+const GAUGE_MARGIN = 16;
+const LABEL_WIDTH = 36;
+const LABEL_HEIGHT = 16;
+const LABEL_OFFSET = 4;
+const TITLE_OFFSET = 11;
+const EXTRA_NEEDLE_LENGTH = 4;
+export const ARC_PADDING = 2;
+export const BREAKPOINTS = [52, 70, 88, 106, 124, 142];
```

Comment:
what is breakpoints.

Reply:
As per the design doc, the font size of the chart value and the arc width varies depending on the gauge chart size. We can't use the CSS media queries here because the styles are conditionally applied based on browser and operating system parameters. So I created these breakpoints to update styles according to the outer radius of the gauge chart.